File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    
    for c in commands:
        i = c[0]-1
        j= c[1]+1
        k = c[2]-1
        # 1 & 2
        tmp = array[i:j-1]
        tmp=sorted(tmp)
        
        answer.append(tmp[k])
        
        
        
        
        
        
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        for(int i=0; i<commands.length; i++){
            int[] cur =commands[i];
            int[] subArr = Arrays.copyOfRange(array, cur[0]-1,cur[1]);
            Arrays.sort(subArr);
            
            answer[i]=subArr[cur[2]-1];
        }
        
        return answer;
    }
}